Source/include/Ably/ARTPush.h

```diff
@@ -33,6 +33,13 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)didAblyPushRegistrationFail:(nullable ARTErrorInfo *)error;
 
+/**
+ Ably will call the implementation of this method when the activation process is completed with success, so you can request other types of device tokens, such as `location`, `pushtotalk` etc.
+ See https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/sending_notification_requests_to_apns for possible values of `apns-push-type`.
+ Before making request to Apple for alternative device token you should check whether you've already did this after application launch.
+ */
+- (void)shouldRequestAlternativeDeviceToken;
```

Comment:
`didActivateAblyPush:` says that device token was saved on the Ably server, hence "Ably" in the name (I think). This is not the case for this method.

Reply:
Hmm, that's one way of looking at it, but I saw it as just relating to the fact that it came from the Ably SDK — it also provides some namespacing to avoid clashing with other methods in a user’s class.

Reply:
https://github.com/ably/ably-cocoa/pull/1771#discussion_r1264776018